server/src/main/scala/com/cloudera/livy/utils/LivySparkUtils.scala

```diff
@@ -89,27 +100,54 @@ object LivySparkUtils {
     val exitCode = process.waitFor()
     val output = process.inputIterator.mkString("\n")
 
-    val regex = """version (.*)""".r.unanchored
-
+    var sparkVersion = ""
     output match {
-      case regex(version) => version
+      case sparkVersionRegex(version) => sparkVersion = version
       case _ =>
         throw new IOException(f"Unable to determine spark-submit version [$exitCode]:\n$output")
     }
+
+    var scalaVersion = ""
+    output match {
+      case scalaVersionRegex(version) => scalaVersion = version
+      case _ =>
+    }
+
+    (sparkVersion, scalaVersion)
   }
 
   /**
-    * Return formatted Spark version.
-    * @param version Spark version
-    * @return Two element tuple, one is major version and the other is minor version
-    */
+   * Return formatted Spark version.
+   *
+   * @param version Spark version
+   * @return Two element tuple, one is major version and the other is minor version
+   */
   def formatSparkVersion(version: String): (Int, Int) = {
-    val versionPattern = """(\d)+\.(\d)+(?:\.\d*)?""".r
+    val versionPattern = """(\d)+\.(\d)+.*""".r
     version match {
       case versionPattern(major, minor) =>
         (major.toInt, minor.toInt)
       case _ =>
         throw new IllegalArgumentException(s"Fail to parse Spark version from $version")
     }
   }
+
+  /**
+   * Return formatted Scala version, if it cannot be parsed from input version string, it will
+   * pick default Scala version related to Spark version.
+   *
+   * @param version Scala version String
+   * @param sparkVersion formatted Spark version.
+   * @return Tuple of Scala version, one is major version and the other is minor version.
+   */
+  def formatScalaVersion(version: String, sparkVersion: (Int, Int)): (Int, Int) = {
```

Comment:
I think we can just handle Scala version as a string and do simple string matching here. As we are not using the Scala version number to do comparsion. Also since we support only "2.10" and "2.11", there's no much combinations.